From: Felix Fietkau <nbd@nbd.name>
Date: Sat, 9 Aug 2025 15:55:11 +0200
Subject: [PATCH] ucode: adjust for new ext resource type

ucv_resource_dataptr always returns null for resources with embedded data.

Signed-off-by: Felix Fietkau <nbd@nbd.name>
---

--- a/ucode.c
+++ b/ucode.c
@@ -449,7 +449,7 @@ rpc_ucode_script_call(struct ubus_contex
 		res = uc_vm_stack_pop(&script->vm);
 
 		/* The handler function invoked a nested aync ubus request and returned it */
-		if (ucv_resource_dataptr(res, "ubus.deferred")) {
+		if (ucv_resource_data(res, "ubus.deferred")) {
 			/* Install guard timer in case the reply callback is never called */
 			callctx->timeout.cb = rpc_ucode_request_timeout;
 			uloop_timeout_set(&callctx->timeout, request_timeout);
